### ISSUE TITLE
Allow disabling Shared History

### ISSUE THREAD

**Issue by AGlezB** [a85/POSTMan-Chrome-Extension#8246]
**Is your feature request related to a problem? Please describe.**
Postman deskto app nagging about monthly usage limits because of Shared History

**Describe the solution you'd like**
A way to completely disable shared history.

**Describe alternatives you've considered**
Clear all every time the nagging starts.

**Additional context**
Having hidtory is great as long as it behaves the same as in browsers:
1. You can disable it.
2. You can configure how long is kept.
3. It doesn't bother you about how much history you have.

I understand # 3 cannot go away so please give me # 1 and hopefully # 2 as well.

**Comment by sankalp0o** [comment]
Hey @AGlezB 

Postman only notifies you about history in team workspaces. One way to "disable" this is to switch to a personal workspace (from the header bar in the app) where the history you create isn't shared with your team and hence you won't be running into the limit of shared history.

I'm closing this issue but please feel free to open it again if this doesn't solve your problem or if you have any other suggestions on how we can make the experience better!

**Comment by AGlezB** [comment]
> 
> 
> Hey @AGlezB
> 
> Postman only notifies you about history in team workspaces. One way to "disable" this is to switch to a personal workspace (from the header bar in the app) where the history you create isn't shared with your team and hence you won't be running into the limit of shared history.

Great! Except Shared History increases automatically and unlike other "usage" counters it doesn't hurt your service to disable it because it doesn't add anything to it.

@sankalp0o Maybe you can explain to me why Shared History is so useful and important that you closed the issue without a second thought.

**Comment by sankalp0o** [comment]
@AGlezB I apologize if my responses sounded unthoughtful. I meant to suggest that shared history is an optional feature and you can continue using personal workspaces if the feature is not useful. Can I understand your use case more? Perhaps that will help me figure out ways to help you.

I'm also not sure what you meant when you said that the shared history "increases automatically". Can you help me understand this as well?

**Comment by AGlezB** [comment]
I'll try to answer your questions one by one.

> I meant to suggest that shared history is an optional feature and you can continue using personal workspaces if the feature is not useful.

As of today if I choose to use a Team Workspace I am _forced_ to have Shared History (SH). I cannot disable SH for team workspaces (TW) in the same way I cannot enable SH for personal workspaces (PW). That means the only way not to use SH is --as you pointed out-- to use PW thus loosing the advantages of TWs.

> I'm also not sure what you meant when you said that the shared history "increases automatically". Can you help me understand this as well?

This is what Postman tracks as _usage_:
- APIs
- Integrations
- Mock Usage
- Monitoring Usage
- Postman API Usage
- Public Documentation Views
- Shared History
- Shared Requests

Those can be categorized as follows:

1. The ones that only increase manually. Can decrease by deleting them.
- APIs
- Integrations
- Shared Requests

2. The ones that increase automatically but have to be used/enabled manually. Always increase and reset periodically.
- Mock Usage
- Monitoring Usage
- Postman API Usage
- Public Documentation Views

3. The single one that increases by using a different feature --a core feature-- and you cannot disable  it or control it except for clearing all or deleting specific entries when the nagging starts.
- Shared History

Groups 1 and 2 requiere that I manually add requests or integrations or use mockups or monitor usage. If I don't use a feature it will not increase.

SH on the other increases when you run requests --the core feature I mentioned before-- in TWs regardless of wheter you want it or not. Then it will begin to nag about upgrading because you've hit the usage limit. At that point SH stops being a feature and becomes a bug.

I'm not asking for SH to be removed since I expect someone will have a use for it.
I'm asking for a simple switch to disable SH at the workspace level and a global switch to disable SH in my account. That is all.

This is our use case:
One of our clients asks that we integrate their existing Product Management Solution in C# with Magento REST API so they can publish their products from their existint solution with one click. Since Magento docs... leave a lot to be desired we fire Postman, create a TW and a few requests to see what the actual responses are. Then we add a few examples to each request and forget about it until until the API changes or we have to make adjoustments to the integration.
SH does not fit anyware in our use case since we use examples to snapshot the requests and responses we care about.

I hope this answer your questions. Please feel free to ask about any particular I didn't manage to explain properly.

**Comment by a85** [comment]
@AGlezB Thank you for the explanation. However, the shared history feature works as designed. Shared history limits and shared request limits exist for people to try out the feature and see if there is value in collaborating and eventually buying the paid product which support the ability to develop the free product. If you are not ready to pay or don't want to pay then we recommend using the free version.

While in some narrow use cases like the one that you described there are instances where you can continue sharing, adding additional controls to make certain things disabled is not of our roadmap and will not be something that we will be able to devote bandwidth to. These controls complicate the roadmap, eat up engineering bandwidth, and reduce product quality at the expense of customers who actually do want to pay for the product.

As @sankalp0o mentioned you can use personal workspaces. You can also potentially export/import and replicate a workflow. I am closing this issue for now.

**Comment by AGlezB** [comment]
> Shared history limits and shared request limits exist for people to try out the feature and see if there is value in collaborating and eventually buying the paid product

@a85 If you look at your [pricing](https://www.postman.com/pricing) there is no mention whatsoever to Shared History so there is no guarantee that I will get rid of the nagging regardless of how much I pay for your services, because we will _always_ end up hitting the quota, because you add to SH every single time I execute a request while completely ignoring if I want the _feature_ or not.

If you had bothered to mention somewere --anywhere-- that we will not have a SH quota in paid plans I would not have opened the issue.

> is not of our roadmap and will not be something that we will be able to devote bandwidth to.

I understand.

> These controls complicate the roadmap, eat up engineering bandwidth, and reduce product quality at the expense of customers who actually do want to pay for the product.

I agree. Thoughtless _features_ often do so.
I would love to see you SH analytics. I'm sure that would be enlightening.

Thanks anyway.
@sankalp0o I do appreciate that you took the time to give this a second thought.

**Comment by a85** [comment]
There is no limit on shared requests - either history or collections - on all plans. It is mentioned under the 2nd row in the pricing page. "Unlimited workspaces and shared requests".  :)

**Comment by AGlezB** [comment]
@a85 My bad. Thanks.

**Comment by a85** [comment]
@AGlezB No problem. Appreciate the feedback and the team is listening constantly. Also, we'll see if we can make the pricing page copy clearer.

**Comment by edamreed** [comment]
Hi, I agree with the original reporter.
The Free Team features are great (I'm not ungrateful) but I have to keep deleting the History as it quickly reaches the limit (I'm running the same requests over and over).
Not having the option for me to switch off Shared History on a Team Workspace is a pain.

I want to share the request definitions amongst the team, I don't want/need to share the History.
Having the option for me to stop my requests going in to the shared history would be a good addition.

**Comment by blockchainforger** [comment]
Hi. I am using the TEAM plan, for a team of 11 members. 

We uses different workspace, to group different projects. Most developers switch workspace each month, and here's the issue.

1. The shared history is so huge, it takes a very long time to sync the changes. The only thing we care about are new requests, or modified requests. Not the history.
2. Deleting the history, with the button "Delete All" does not work. It just keeps running and fail at some point.

I had to delete the history, individually, by deleting each day's worth of history. I did this manually to delete a 2 years of history. Took me an hour to do so.

Then another problem occur, when the developers switched to the workspace, it is now syncing the "deleted history". The whole postman just hangs and they are unable to do anything for an hour or so.

Now, we implemented a rule to delete the history every day. Which is really a pain.

I would love to have a way to disable shared history. It is more of a cons to my team.
